flink-runtime/src/main/java/org/apache/flink/runtime/util/config/memory/ProcessMemoryUtils.java

```diff
@@ -158,6 +156,25 @@ public JvmMetaspaceAndOverhead deriveJvmMetaspaceAndOverheadFromTotalFlinkMemory
 		return jvmMetaspaceAndOverhead;
 	}
 
+	private MemorySize deriveJvmOverheadFromTotalFlinkMemoryAndOtherComponents(
+			Configuration config,
+			MemorySize totalFlinkMemorySize) {
+		MemorySize totalProcessMemorySize = getMemorySizeFromConfig(config, options.getTotalProcessMemoryOption());
+		MemorySize jvmMetaspaceSize = getMemorySizeFromConfig(config, options.getJvmOptions().getJvmMetaspaceOption());
+		MemorySize totalFlinkAndJvmMetaspaceSize = totalFlinkMemorySize.add(jvmMetaspaceSize);
```

Comment:
It seems as if we are doing these computations twice. Once here and in `deriveJvmMetaspaceAndOverheadFromTotalFlinkMemory` before this method here is called.